src/type-checker.cc

```diff
@@ -524,7 +519,7 @@ Result TypeChecker::OnIndexedFuncRef(Index* out_index) {
   Type type;
   Result result = PeekType(0, &type);
   if (!type.IsReferenceWithIndex()) {
-    type = Type::Reference;
+    type = Type(Type::Reference, kInvalidIndex);
```

Comment:
This was inconsistent in the code, sometimes `Type(Type::Reference, kInvalidIndex)` is used, sometimes just a `Type::Reference`. Now we must use the `Type(Type::Reference, kInvalidIndex)` form.

Reply:
sounds good!